tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1198,17 +1200,18 @@ ROOT::Experimental::Internal::RNTupleFileWriter::Append(std::string_view ntupleN
    assert(file->IsBinary());
 
    auto writer = std::unique_ptr<RNTupleFileWriter>(new RNTupleFileWriter(ntupleName, maxKeySize));
-   writer->fFileProper.fDirectory = &fileOrDirectory;
+   writer->fFile.emplace<RFileProper>(&fileOrDirectory);
```

Comment:
I think I would prefer simplicity and have `RFileProper` also default-constructible:
```suggestion
   auto &fileProper = writer->fFile.emplace<RFileProper>();
   fileProper.fDirectory = &fileOrDirectory;
```